include/deal.II/base/mpi.h

```diff
@@ -1539,11 +1539,195 @@ namespace Utilities
     }
 
 
+    template <typename T>
+    T
+    _builtin_broadcast(const MPI_Comm    &comm,
+                       const T           &object_to_send,
```

Comment:
I think we also want to allow arrays of supported types. I would rather go with `ArrayView<T>` here.

Reply:
Regarding the implementation of `_builtin_brodcast` with `ArrayView`, how should I communicate the size of the array between the processes? Can I assume that the buffer (with the appropriate size) is already allocated in client processes?

Reply:
Good question! I think the `ArrayView` version requires some more discussion (if we decide that we want it). It would likely make more sense to let the user provide the target as a parameter and not as return value.